### DISCUSSION TITLE
Checks Annotation Doesn't Go to File/Show File

### DISCUSSION THREAD

**Discussion by adzienis** [community/community#15412] [Actions]
Somewhat related to https://github.com/github/feedback/discussions/9144, but when adding an annotation, the link unfortunately is unable to correctly route to the offending file:
![Screen Shot 2022-04-22 at 1 44 33 PM](https://user-images.githubusercontent.com/7027362/164766830-95a8b959-3b7d-44c2-8f00-a127d57ec5ec.png)

I have an action that can annotate and indicate which tests are failing based on the line number and path. Unfortunately, when clicking the link in the example screenshot for the check, it is unable to go to the file, and shows the specific commit the check was run on (which did not include the failing file/test). This is most definitely because Github currently? doesn't annotate files not changed by the commit. I see references to a Beta option of a section in the PR view of `Unchanged files with check annotations`, but can never get one to show/figure out where to turn it on.

I am wondering if there is an alternate usage flow for other users of the Checks API? I just think it would be helpful to annotate unchanged files (for example the tests), if a change in application code causes a test to fail.

Thanks in advance for any help/follow ups, and I thoroughly enjoy the complete feature set of Github Actions. It's great!

**Comment by cryptoguardian-it** [comment]
following...

**Comment by ohookins** [comment]
I've been experiencing this as well. I can see how it could be intentional, since the file where the error was found was not changed in the commit. However, there are numerous situations where we might change some code that has knock-on effects in other parts of the codebase and it wouldn't make sense to not be able to link to these just because the particular line wasn't changed. I know there has been a historical limitation of not being able to comment on lines not changed in a PR - not sure if the checks API has the same limitation?

But would appreciate this being fixed!
